pyflakes/checker.py

```diff
@@ -964,14 +980,27 @@ def IMPORT(self, node):
 
     def IMPORTFROM(self, node):
         if node.module == '__future__':
-            if not self.futuresAllowed:
+            def has_only_future_imports():
```

Comment:
I'm still not sure why we need this, instead of just making the logic above, around line 632, set `_futures_allowed` to False as soon as it sees any kind of import that is not `from __future__ ...`. I think it would be much simpler to understand the logic were it in just one place.